src/utils/json/flatten.rs

```diff
@@ -273,52 +274,83 @@ pub fn flatten_array_objects(
+            .collect()),
+        Value::Object(map) => {
+            let results = map
+                .iter()
+                .fold(vec![Map::new()], |results, (key, val)| match val {
+                    Value::Array(arr) => arr
+                        .iter()
+                        .flat_map(|flatten_item| flatten_json(flatten_item).unwrap_or_default())
+                        .flat_map(|flattened_item| {
+                            results.iter().map(move |result| {
+                                let mut new_obj = result.clone();
+                                new_obj.insert(key.clone(), flattened_item.clone());
+                                new_obj
+                            })
+                        })
+                        .collect(),
+                    Value::Object(_) => flatten_json(val)
+                        .unwrap_or_default()
+                        .iter()
+                        .flat_map(|nested_result| {
+                            results.iter().map(move |result| {
+                                let mut new_obj = result.clone();
+                                new_obj.insert(key.clone(), nested_result.clone());
+                                new_obj
+                            })
                         })
-                    })
-                    .collect(),
-                Value::Object(_) => flatten_json(val)
-                    .iter()
-                    .flat_map(|nested_result| {
-                        results.iter().map(move |result| {
-                            let mut new_obj = result.clone();
-                            new_obj.insert(key.clone(), nested_result.clone());
-                            new_obj
+                        .collect(),
+                    _ => results
+                        .into_iter()
+                        .map(|mut result| {
+                            result.insert(key.clone(), val.clone());
+                            result
                         })
-                    })
-                    .collect(),
-                _ => results
-                    .into_iter()
-                    .map(|mut result| {
-                        result.insert(key.clone(), val.clone());
-                        result
-                    })
-                    .collect(),
-            })
-            .into_iter()
-            .map(Value::Object)
-            .collect(),
-        _ => vec![value.clone()],
+                        .collect(),
+                });
+
+            Ok(results.into_iter().map(Value::Object).collect())
+        }
+        _ => Ok(vec![value.clone()]),
+    }
+}
+
+/// recursively checks the level of nesting for the serde Value
+/// if Value has more than 4 levels of hierarchy, returns true
+/// example -
+/// 1. `{"a":{"b":{"c":{"d":{"e":["a","b"]}}}}}` ~> returns true
+/// 2. `{"a": [{"b": 1}, {"c": 2}], "d": {"e": 4}}` ~> returns false
+fn has_more_than_four_levels(value: &Value, current_level: usize) -> bool {
```

Comment:
This would just add compute cycles to ingestion without helping in any other way, do we need this?